machine-runner/tests/esm/runner.test.ts

```diff
@@ -353,7 +352,83 @@ describe('machine as async generator', () => {
     machine.destroy()
   })
 
-  it('should resolve all previously unsolved yielded promises on one caughtUp event', async () => {
+  it('should only yield on "next" event', async () => {
+    const watchPromise = (promise: Promise<unknown>) => {
+      let finished = false
+      const isFinished = () => finished
+      promise.finally(() => (finished = true))
+      return { isFinished }
+    }
+
+    const r1 = new Runner(On, { toggleCount: 0 })
+    const { machine } = r1
+
+    const issueCommand = (snap: StateOpaque.Of<typeof machine> | null) => {
+      const state = snap?.as(On) || snap?.as(Off)
+      return state?.commands?.toggle()
+    }
+
+    const promise1 = machine.next()
+    const promise1Watcher = watchPromise(promise1)
+
+    expect(promise1Watcher.isFinished()).toBe(false)
+    await r1.feed([], true)
+    expect(promise1Watcher.isFinished()).toBe(true) // true because this is the first caughtUp
+
+    const promise2 = machine.next()
+    const promise2Watcher = watchPromise(promise2)
+    expect(promise2Watcher.isFinished()).toBe(false)
+    await r1.feed([], true)
+    expect(promise2Watcher.isFinished()).toBe(false) // false because there is no change
+
+    await issueCommand(machine.get())
```

Comment:
this code relies on event loop details — perhaps we can make it more obvious using the following before asserting `.isFinished()` values:

```typescript
await new Promise((res) => setTimeout(res, 1))
```

Reply:
Thanks for the catch